src/sage/arith/misc.py

```diff
@@ -2044,19 +2077,45 @@ def xgcd(a, b):
         sage: S.<y> = R.fraction_field()[]
         sage: xgcd(y^2, a*h*y + b)
         (1, 7*a^2/b^2, (((-h)*a)/b^2)*y + 1/b)
+
+    Tests with randomly generated integers::
+
+        sage: import numpy as np
+        sage: N, M = 1000, 10000
+        sage: a = np.random.randint(M, size=N) * np.random.randint(M)
+        sage: r = xgcd(a)
+        sage: len(r) == len(a) + 1
+        True
+        sage: r[0] == gcd(a)
+        True
+        sage: sum(c * x for c, x in zip(r[1:], a)) == gcd(a)
+        True
     """
-    try:
+    if b is not None:
+        # xgcd of two elements
+        try:
+            return a.xgcd(b)
+        except AttributeError:
+            a = py_scalar_to_element(a)
+            b = py_scalar_to_element(b)
+        except TypeError:
+            b = py_scalar_to_element(b)
         return a.xgcd(b)
-    except AttributeError:
-        a = py_scalar_to_element(a)
-        b = py_scalar_to_element(b)
-    except TypeError:
-        b = py_scalar_to_element(b)
-    return a.xgcd(b)
+
+    # xgcd for several elements (possibly more than one)
+    if len(a) == 0:
+        return (ZZ(0),)
+    a = Sequence(a, use_sage_types=True)
+    res = [a.universe().zero()]
+    for b in a:
+        g, s, t = xgcd(res[0], b)
+        res = [g] + [s * c for c in res[1:]] + [t]
```

Comment:
```suggestion
        res[0] = g
        for i in range(1, len(res)):
            res[i] *= s
        res.append(t)
```
Should be much faster :)